### PR TITLE
Add live and current variant to AllEvents query and their documentation

### LINKED ISSUE #4518: Akka.Persistence.Query: need to make it clear that PersistenceIds queries only work on what the current JournalActor has observed.
Version: 1.4.8

Common issue that I run into all the time myself: Akka.Persistence.Query `PersistenceIds` or `CurrentPersistenceIds` queries _only query what actually passes through the journal actors on their live, running nodes locally_ - not on the entire content of everything stored inside the database.

Need to make that clear going forward.

### PR DESCRIPTION
**Pull request by Arkatufus**
Continuation of #4521 
Closes #4518

Updates:
- The previous PR lacks the live and current variants.
- Add `AllEvents` and `CurrentAllEvents` documentation
- Updates `AllPersistenceIds` and `PersistenceIds` documentation

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@Arkatufus still waiting on docs for this, right?

**Comment by Arkatufus** [comment]
yes, documentation should be in now

**Comment by Aaronontheweb** [comment]
@Arkatufus looks like we have a compilation error here

File: src/contrib/persistence/Akka.Persistence.Query.Sql/AllEventsPublisher.cs [hunk 1]
```diff
@@ -77,76 +88,129 @@ private bool Idle(object message)
...
+    internal sealed class LiveAllEventsPublisher : AbstractAllEventsPublisher
+    {
+        private readonly ICancelable _tickCancelable;
+        public LiveAllEventsPublisher(long fromOffset, TimeSpan refreshInterval, int maxBufferSize, string writeJournalPluginId)
+            : base(fromOffset, maxBufferSize, writeJournalPluginId)
+        {
+            _tickCancelable = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(refreshInterval, refreshInterval, Self, EventsByTagPublisher.Continue.Instance, Self);
+        }
+
+        protected override long ToOffset => long.MaxValue;
```
Comment (Aaronontheweb) [inline src/contrib/persistence/Akka.Persistence.Query.Sql/AllEventsPublisher.cs:147]: LGTM - need this to make `IIsTimeForReplay` function correctly on live queries

File: src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs [hunk 2]
```diff
@@ -37,42 +33,35 @@ public void ReadJournal_should_implement_IAllEventsQuery()
         }
 
         [Fact]
-        public virtual void ReadJournal_query_AllEvents_should_find_existing_events()
+        public virtual void ReadJournal_query_AllEvents_should_find_new_events()
```
Comment (Aaronontheweb) [inline src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs:36]: LGTM

File: src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs [hunk 3]
```diff
@@ -81,54 +70,42 @@ public virtual void ReadJournal_query_AllEvents_should_find_events_from_offset_e
...
-            {
-                var idx = i + 1;
-                probe.ExpectNext<EventEnvelope>(p =>
-                    p.PersistenceId == "a" && p.SequenceNr == idx && p.Event.Equals("a green apple"));
-                Output.WriteLine(idx.ToString());
-            }
-
-            probe.ExpectComplete();
-            probe.ExpectNoMsg(TimeSpan.FromMilliseconds(500));
+            // everything before "streaming" are not included, since exclusive offset
```
Comment (Aaronontheweb) [inline src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs:104]: LGTM

**Pull request merged**